extensions/amp-vk/0.1/amp-vk.js

```diff
@@ -0,0 +1,249 @@
+      queryParams = {
+        'app': this.apiId_,
+        'width': '100%',
+        '_ver': 1,
+        'poll_id': this.pollId_,
+        'amp': 1,
+        'url': pageUrl,
+        'title': 'AMP Poll',
+        'description': '',
+        'referrer': pageReferrer,
+      };
+    }
+
+    for (const param in queryParams) {
+      q += `&${param}=${encodeURIComponent(queryParams[param])}`;
+    }
+
+    src += `?${q.substr(1)}&${createdTime}`;
+
+    return src;
+  }
+
+  /** @override */
+  buildCallback() {
+    user().assert(this.element.getAttribute('data-embedtype'),
+        'The data-embedtype attribute is required for <amp-vk> %s',
+        this.element);
+
+    this.embedType_ = this.element.getAttribute('data-embedtype');
+
+    if (this.embedType_ === embedTypes.POST) {
+      this.postBuildCallback_();
+    } else if (this.embedType_ === embedTypes.POLL) {
+      this.pollBuildCallback_();
+    }
+  }
+
+  /** @private */
+  postBuildCallback_() {
+    user().assert(this.element.getAttribute('data-hash'),
+        'The data-hash attribute is required for <amp-vk> Post %s',
+        this.element);
+
+    user().assert(this.element.getAttribute('data-owner_id'),
+        'The data-owner_id attribute is required for <amp-vk> Post %s',
+        this.element);
+
+    user().assert(this.element.getAttribute('data-post_id'),
+        'The data-post_id attribute is required for <amp-vk> Post %s',
+        this.element);
+
+    this.ownerId_ = this.element.getAttribute('data-owner_id');
+    this.postId_ = this.element.getAttribute('data-post_id');
+    this.hash_ = this.element.getAttribute('data-hash');
+  }
+
+  /** @private */
+  pollBuildCallback_() {
+    user().assert(this.element.getAttribute('data-api_id'),
+        'The data-api_id attribute is required for <amp-vk> Poll %s',
+        this.element);
+
+    user().assert(this.element.getAttribute('data-poll_id'),
+        'The data-poll_id attribute is required for <amp-vk> Poll %s',
+        this.element);
+    this.pollId_ = this.element.getAttribute('data-poll_id');
+    this.apiId_ = this.element.getAttribute('data-api_id');
+  }
+
+  /** @override */
+  layoutCallback() {
+    const iframe = this.element.ownerDocument.createElement('iframe');
+    this.iframe_ = iframe;
+
+    this.unlistenMessage_ = listen(
+        this.win,
+        'message',
+        this.handleVkIframeMessage_.bind(this)
+    );
+
+    iframe.src = this.getIFrameSrc_();
+    iframe.setAttribute('name', 'fXD');
+    iframe.setAttribute('scrolling', 'no');
+    iframe.setAttribute('frameborder', '0');
+
+    this.applyFillContent(iframe);
+    this.element.appendChild(iframe);
+
+    return this.iframePromise_ = this.loadPromise(iframe);
+  }
+
+  /**
+   * @param {!Event} e
+   * @private
+   */
+  handleVkIframeMessage_(e) {
+    if (e.origin !== 'https://vk.com' ||
+        e.source !== this.iframe_.contentWindow) {
+      return;
+    }
+    const eventData = getData(e);
```

Comment:
based on https://ampvk.herokuapp.com/examples/vk.amp.html, I have noticed the page constantly sends `resize` messages without ever stopping. please fix.

Reply:
Thanks for reporting - raised with vk.com team, will fix